### PR TITLE
feat: Percy visual testing

### PR DESCRIPTION
**Pull request by janpio**
Quick test with Percy to make PRs like https://github.com/prisma/docs/pull/1995 simpler to test and confirm.

### CODE REVIEW

**Comment by janpio** [comment]
Unfortunately getting the snapshots on Netlify takes quite a while:
```
9:45:46 PM: [percy] created build #9: https://percy.io/6faef5e0/prisma-docs/builds/11449855
9:45:47 PM: [percy] percy has started.
...
9:51:38 PM: [percy] waiting for 245 snapshots to complete...
9:51:38 PM: [percy] done.
9:51:38 PM: [percy] finalized build #9: https://percy.io/6faef5e0/prisma-docs/builds/11449855
```

6 minutes additional build time is to much.

So we probably would want to extract that into a GH Actions run here that is optimized to do that I would say.

**Comment by janpio** [comment]
And another problem: 
Running Percy on Netlify creates _very_ different results from running it on Gitpod (a dev environment):

image

**Comment by janpio** [comment]
That was quick: 
image
Blocked until someone buys a paid plan for us internally 🤷

**Comment by avocadowastaken** [comment]
> 6 minutes additional build time is to much.
> 
> So we probably would want to extract that into a GH Actions run here that is optimized to do that I would say.

Yeah, looks like Netlify is not optimized for that

> And another problem:
> Running Percy on Netlify creates _very_ different results from running it on Gitpod (a dev environment):

Looks like it uses different fonts, I had same problem with loki js, main reason why I started to use Percy. Strange that it occurs on Percy too. Because the mechanic they use to produce the snapshots should be consistent.

> Blocked until someone buys a paid plan for us internally 🤷

I thought they provide free plans for the OSS  https://www.browserstack.com/open-source

**Comment by janpio** [comment]
> > And another problem:
> > Running Percy on Netlify creates _very_ different results from running it on Gitpod (a dev environment):
> 
> Looks like it uses different fonts, I had same problem with loki js, main reason why I started to use Percy. Strange that it occurs on Percy too. Because the mechanic they use to produce the snapshots should be consistent.

Different fonts and missing code block etc - so possibly some code not executed. Let's hope it is not the case via GH Actions.


> > Blocked until someone buys a paid plan for us internally 🤷
> 
> I thought they provide free plans for the OSS [browserstack.com/open-source](https://www.browserstack.com/open-source)

The money is not the problem, the weekend is 😆 Applied anyway, maybe Browserstack is still working and gets back to us sooner.

**Comment by janpio** [comment]
> > I thought they provide free plans for the OSS [browserstack.com/open-source](https://www.browserstack.com/open-source)
> 
> The money is not the problem, the weekend is 😆 Applied anyway, maybe Browserstack is still working and gets back to us sooner.

They do it turns out (suuuuuper fast!), but Percy is not included :/

**Comment by janpio** [comment]
Update: Confirmed they only include the other services, not Percy. Need to wait for the credit card next week.

**Comment by avocadowastaken** [comment]
@janpio btw did you add github integration? it should add Percy github check status https://docs.percy.io/docs/github#step-3-run-and-review-builds

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  gatsby-for-percy:
+    name: Percy screenshots via Gatsby build
+    runs-on: ubuntu-latest
+    env:
+      PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - run: yarn
+      - run: yarn gatsby build
```
Comment (avocadowastaken) [inline .github/workflows/ci.yml]: I've copied this file in #2020, so there might be merge conflicts, also should it be
```suggestion
      - uses: actions/setup-node@v2
         with:
           cache: 'npm'
           node-version: '14'
      - run: npm ci
      - run: npm run build
```

**Comment by janpio** [comment]
Yes, but I think it will only pop up after a successful run - which we did not have yet as we are missing payment data. But we are working on that now.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (janpio) [reply .github/workflows/ci.yml]: Oh lol, why are we not using yarn here... Thanks for the suggestion.

**Comment by janpio** [comment]
Here is the Percy GitHub check and it should also be public. If that works and makes sense, we can merge this to the main branch and you can rebase your PRs and then we should be able to visually see if anything changed.

**Comment by janpio** [comment]
Ugh. Unfortunately it looks like it is also using a different font and not rendering code blocks and so on :/ We'll need to debug this before this can be useful :(

**Comment by janpio** [comment]
Actually, merging this anyway as it gives us _some_ visual feedback.

**Comment by janpio** [comment]
Trying that in https://github.com/prisma/docs/pull/2025

**Comment by janpio** [comment]
Update: Nope, "No diffs" with this change.

**Pull request merged**